### PR TITLE
Megahit tool

### PR DESCRIPTION
**Pull request by cmonjeau**
Hi,

I propose a PR for a pretty tool for metagenomics assembly : megahit.

Thanks

(ping @abretaud )

### CODE REVIEW

File: tools/megahit/megahit_wrapper.xml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                <param name="fastq_input1" format="fastq,fastqsanger,fasta" type="data_collection" collection_type="list:paired" label="Select a paired collection"/>
+            </when>
+            <when value="single">
+                <param name="single_files" multiple="true" format="fastq,fastqsanger,fasta" type="data" label="Single-end file(s)" help="FASTQ/FASTA/FASTQ.GZ files accepted" />
+            </when>
+        </conditional>
+        <param name="min_contig_len" argument="--min-contig-len" type="integer" value="200" label="minimum length of contigs to output" />
+        <section name="basic_section" title="Basic assembly options" expanded="False">
+            <param name="min_count" argument="--min-count" type="integer" value="2" label="minimum multiplicity for filtering (k_min+1)-mers"/>
+            <param name="k_min" argument="--k-min" type="integer" value="21" label="minimum kmer size" max="127" help="must be odd number"/>
```
Comment (abretaud) [inline tools/megahit/megahit_wrapper.xml]: I wonder if there's a way to check for odd/even numbers

File: tools/megahit/megahit_wrapper.xml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            </param>
+            <when value="paired">
+                <param name="fastq_input1" format="fastq,fastqsanger,fasta" type="data_collection" collection_type="list:paired" label="Select a paired collection"/>
+            </when>
+            <when value="single">
+                <param name="single_files" multiple="true" format="fastq,fastqsanger,fasta" type="data" label="Single-end file(s)" help="FASTQ/FASTA/FASTQ.GZ files accepted" />
+            </when>
+        </conditional>
+        <param name="min_contig_len" argument="--min-contig-len" type="integer" value="200" label="minimum length of contigs to output" />
+        <section name="basic_section" title="Basic assembly options" expanded="False">
```
Comment (abretaud) [inline tools/megahit/megahit_wrapper.xml]: Is it intended to have it *not* expanded? (as it's written to be basic options)

**Comment by abretaud** [comment]
Thank you Cyrilou,
It's a 👍 for me, but I leave it open a little in case someone else wants to review it

File: tools/megahit/megahit_wrapper.xml (hunk 2 above)
Comment (cmonjeau) [reply tools/megahit/megahit_wrapper.xml]: I moved the param in basic section

File: tools/megahit/megahit_wrapper.xml (hunk 1 above)
Comment (cmonjeau) [reply tools/megahit/megahit_wrapper.xml]: I don't know. I found no examples in iuc tools.

Comment (nsoranzo) [reply tools/megahit/megahit_wrapper.xml]: If you think it's necessary, I think you can use a `<validator type="expression" />`, see https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/parameters/validation.py#L67

**Comment by bgruening** [comment]
`min`, `max`, `arguments`, tests ... this is an awesome contribution! Thanks a bunch @cmonjeau!

**Pull request merged**